homeassistant/components/nomaiq/__init__.py

```diff
@@ -0,0 +1,74 @@
+"""The nomaiq integration."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+import ayla_iot_unofficial
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_CLIENT_ID,
+    CONF_CLIENT_SECRET,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .coordinator import NomaIQDataUpdateCoordinator
+
+_PLATFORMS: list[Platform] = [Platform.LIGHT]
+_LOGGER = logging.getLogger(__name__)
+
+type NomaIQConfigEntry = ConfigEntry[NomaIQDataUpdateCoordinator]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: NomaIQConfigEntry) -> bool:
+    """Set up nomaiq from a config entry."""
+
+    config = entry.data
+    options = entry.options
+
+    client_id = options.get(CONF_CLIENT_ID, config.get(CONF_CLIENT_ID, ""))
+    client_secret = options.get(CONF_CLIENT_SECRET, config.get(CONF_CLIENT_SECRET, ""))
+    username = options.get(CONF_USERNAME, config.get(CONF_USERNAME, ""))
+    password = options.get(CONF_PASSWORD, config.get(CONF_PASSWORD, ""))
```

Comment:
it's stored in one or the other, not in both. and they _should_ be stored in `data`